File: frontend/src/api/health.ts
import { apiClient } from "./client";
import type { HealthResponse } from "../types/health";

export async function fetchHealth() {
  // 最小联调接口，用于确认前端能访问 FastAPI 后端。
  const { data } = await apiClient.get<HealthResponse>("/api/v1/health");
  return data;
}

File: app/api/v1/endpoints/health.py
"""健康检查 API。

部署或本地启动时，可以访问 /api/v1/health 快速确认 FastAPI 是否正常运行。
"""

from fastapi import APIRouter

from app.schemas.health import HealthResponse

router = APIRouter()


@router.get("/health", response_model=HealthResponse)
async def health_check() -> HealthResponse:
    """GET /api/v1/health：返回基础服务状态，用于监控和调试。"""
    return HealthResponse(status="ok", service="gaokao-pilot")
